adr/ADR-44.md

```diff
@@ -0,0 +1,82 @@
+meta-leader and JetStream asset host versions that requires careful consideration to maintain compatibility.
+
+ * The server a client connects to might be a Leafnode on a significantly older release and might not even have 
+   JetStream enabled
+ * The meta-leader could be on a version that does not support a feature and so would not forward the API fields it 
+   is unaware of to the assigned servers
+ * The server hosting an asset might not match the meta-leader and so can't honor the request for a certain 
+   configuration and would silently drop fields
+
+In general our stance is to insist on homogenous cluster versions in the general case but it's not reasonable to expect this
+for leafnode servers nor is it possible during upgrading and other maintenance windows.
+
+Our current approach is to check the connected-server version and project that it's representitive of the cluster as 
+a whole but this is a known incorrect approach especially for Leafnodes as mentioned above.
+
+This ADR specifies a way for the servers to expose some versioning information to help clients improve the 
+compatability story.
+
+# Server-set metadata
+
+In lieu of enabling fine grained API versioning we want to start thinking about asset versioning instead, the server
+should report the version that created an asset and it should report the version that is hosting the asset and 
+potentially expose a asset version.
+
+We'll store this in the existing `metadata` field allowing us to expand this in time, today we propose the following:
+
+| Name                           | Description                                      |
+|--------------------------------|--------------------------------------------------|
+| `_nats.server.version`         | The current server version hosting an asset      |
+| `_nats.server.created.version` | The version server that first created this asset |
+| `_nats.server.created.time`    | The time the asset was first created             |
+
+We intend to store some client hints in here to help us track what client language and version created assets.
+
+As some of these these are dynamic fields tools like Terraform and NACK will need to understand to ignore these fields 
+when doing their remediation loops.
+
+# Offline Assets
+
+Today when an asset cannot be loaded it's simply not loaded. But to improve compatability, user reporting and 
+discovery we want to support a mode where a stream is visible in Stream reports but marked as offline with a reason.
+
+An offline stream should still be reporting, responding to info and more but no messages should be accepted into it 
+and no messages should be delivered to any consumers, messages can't be deleted, configuration cannot be updated - 
+it is offline in every way that would result in a change.  Likewise a compatible offline mode should exist for Consumers. 
+
+The Stream and Consumer state should get new fields `Offline bool` and `OfflineReason string` that should be set for 
```

Comment:
These should also reject if they set on API - or be `X-NATS-OFFLINE` headers. This would remove the data from the model and not complexity the API - clients can turn those to errors?

Reply:
We already reserve `_nats` in another ADR - but yes need wider communication

Reply:
> These should also reject if they set on API - or be X-NATS-OFFLINE headers. This would remove the data from the model and not complexity the API - clients can turn those to errors?

The offline status would be in the stream state, that seems completely fine and in line with current design